mirrord/operator/src/client.rs

```diff
@@ -287,6 +287,7 @@ impl OperatorApi<NoClientCert> {
         reporter: &mut R,
         progress: &P,
         layer_config: &LayerConfig,
+        ci_api_key: Option<&CiApiKey>,
```

Comment:
I think it might be cleaner to have a new method under `OperatorApi<NoClientCert>` such as `with_ci_api_key()` that converts the API directly to a `OperatorApi<PreparedClientCert>` with a given `CiApiKey`.
The method just need to unwrap the certificate, set operator property for telemetry and set the cert in header. Perhaps these can be extracted from the `prepare_client_cert` method to a shared function.